db/api/src/resolver.rs

```diff
@@ -18,24 +18,41 @@ pub trait HoprDbResolverOperations {
 #[async_trait]
 impl HoprDbResolverOperations for HoprDb {
     async fn resolve_packet_key(&self, onchain_key: &Address) -> Result<Option<OffchainPublicKey>> {
```

Comment:
Theoretically this API could disappear and the `OffchainKeyOrAddress` could take a Arc<Cache> reference and resolve dynamically into either `Address` or `OffchainKey`, this creating a sort of "UniversalAddress".